src/mgr/ClusterState.cc

```diff
@@ -70,16 +70,8 @@ void ClusterState::ingest_pgstats(MPGStats *stats)
   Mutex::Locker l(lock);
 
   const int from = stats->get_orig_source().num();
-  bool is_in = false;
-  objecter->with_osdmap([&is_in, from](const OSDMap &osd_map){
-      is_in = osd_map.is_in(from);
-  });
-
-  if (is_in) {
-    pending_inc.update_stat(from, stats->epoch, std::move(stats->osd_stat));
-  } else {
-    pending_inc.update_stat(from, stats->epoch, osd_stat_t());
-  }
+
+  pending_inc.update_stat(from, stats->epoch, std::move(stats->osd_stat));
```

Comment:
Agree, although it mirrors PGMonitor's behavior, and the commit that added it was one of Sam's that adds a bunch of checking and infrastructure to ensure that it is always in sync with osd_epochs (which AFAICS is obsolete).  In any case, I think it's better this way but I don't want to backport this part to luminous.